src/python_testing/matter_testing_support.py

```diff
@@ -949,12 +954,11 @@ def on_fail(self, record):
 
             record is of type TestResultRecord
         '''
+        self.failed = True
         if self.runner_hook and not self.is_commissioning:
             exception = record.termination_signal.exception
             step_duration = (datetime.now(timezone.utc) - self.step_start_time) / timedelta(microseconds=1)
-            # This isn't QUITE the test duration because the commissioning is handled separately, but it's clsoe enough for now
-            # This is already given in milliseconds
-            test_duration = record.end_time - record.begin_time
+            test_duration = datetime.now(timezone.utc) - self.test_start_time
```

Comment:
When trying to run this on TH we got an error because TH expects an int for duration. Could you convert it to an int representing the time in milliseconds?

Reply:
changed.